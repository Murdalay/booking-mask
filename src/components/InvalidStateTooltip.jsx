import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Icon from 'components/Icon';
import Floater from 'react-floater';

export const TooltipContainer = styled.div`
  display: flex;
  align-items: center;
  font-size: 1.2rem;

  span {
    padding-right: 1rem;
  }
`;

const InvalidStateTooltip = ({ isOpen, target }) => (
  <Floater
    content={
      <TooltipContainer>
        <Icon name="exclamation-circle" width={16} />
        <div>Please fill out this field.</div>
      </TooltipContainer>
    }
    placement="top"
    open={isOpen}
    target={`#${target}`}
  />
);

InvalidStateTooltip.propTypes = {
  isOpen: PropTypes.bool,
  target: PropTypes.string.isRequired,
};

export default InvalidStateTooltip;
