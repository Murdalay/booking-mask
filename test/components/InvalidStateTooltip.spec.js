import React from 'react';

import InvalidStateTooltip from 'components/InvalidStateTooltip';

describe('InvalidStateTooltip', () => {
  const wrapper = mount(<InvalidStateTooltip isOpen={true} target="foobar" />);

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
