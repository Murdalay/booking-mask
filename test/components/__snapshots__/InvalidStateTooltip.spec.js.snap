// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidStateTooltip should render properly 1`] = `
.c1 {
  display: inline-block;
  line-height: 0;
}

.c1 svg {
  height: auto;
  max-height: 100%;
  width: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.2rem;
}

.c0 span {
  padding-right: 1rem;
}

<span>
  <Portal
    containerInfo={
      .c1 {
  display: inline-block;
  line-height: 0;
}

.c1 svg {
  height: auto;
  max-height: 100%;
  width: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.2rem;
}

.c0 span {
  padding-right: 1rem;
}

<div
        style="z-index: 100;"
      >
        <div
          class="__floater"
          style="display: inline-block; filter: drop-shadow(0 0 3px rgba(0, 0, 0, 0.3)); max-width: 300px; opacity: 0; position: relative; visibility: hidden; z-index: 100; padding: 0px 0px 16px;"
        >
          <div
            class="__floater__body"
          >
            <div
              class="__floater__container"
              style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); min-height: 60px; min-width: 200px; padding: 20px; position: relative;"
            >
              <div
                class="c0"
              >
                <span
                  class="isvg loading c1"
                />
                <div>
                  Please fill out this field.
                </div>
              </div>
            </div>
            <div
              class="__floater__arrow"
              style="position: absolute; bottom: 0px; left: 0px; right: 0px; height: 16px;"
            >
              <span
                style="display: inline-flex; position: absolute; bottom: 0px;"
              >
                <svg
                  height="16"
                  version="1.1"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polygon
                    fill="#fff"
                    points="0,0 16,16 32,0"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    }
  >
    <div
      className="__floater"
      style={
        Object {
          "display": "inline-block",
          "filter": "drop-shadow(0 0 3px rgba(0, 0, 0, 0.3))",
          "maxWidth": 300,
          "opacity": 0,
          "padding": "0 0 16px",
          "position": "relative",
          "transition": "opacity 0.3s",
          "visibility": "hidden",
          "zIndex": 100,
        }
      }
    >
      <div
        className="__floater__body"
      >
        <div
          className="__floater__container"
          style={
            Object {
              "backgroundColor": "#fff",
              "color": "#666",
              "minHeight": 60,
              "minWidth": 200,
              "padding": 20,
              "position": "relative",
            }
          }
        >
          <TooltipContainer>
            <div
              className="c0"
            >
              <IconWrapper
                src="/media/icons/exclamation-circle.svg"
                style={
                  Object {
                    "fill": undefined,
                  }
                }
                width={16}
              >
                <span
                  className="isvg loading c1"
                  style={
                    Object {
                      "fill": undefined,
                    }
                  }
                />
              </IconWrapper>
              <div>
                Please fill out this field.
              </div>
            </div>
          </TooltipContainer>
        </div>
        <div
          className="__floater__arrow"
          style={
            Object {
              "bottom": 0,
              "height": 16,
              "left": 0,
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <span
            style={
              Object {
                "bottom": 0,
                "display": "inline-flex",
                "position": "absolute",
              }
            }
          >
            <svg
              height={16}
              version="1.1"
              width={32}
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                fill="#fff"
                points="0,0 16,16 32,0"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </Portal>
</span>
`;
